src/azdata.proposed.d.ts

```diff
@@ -413,6 +413,24 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+	/**
+	 * An interface for receiving diagnostic data from a provider.
+	 */
+	export interface Diagnostics {
+		handleErrorCode(errorCode: number, errorMessage: string): Thenable<diagnostics.ErrorDiagnosticsResponse>;
+	}
+
+	export namespace diagnostics {
+		export interface ErrorDiagnosticsResponse {
```

Comment:
Please add doc comment: e.g. "Error diagnostics response containing action to be taken for received error."